src/libs/ws/session.rs

```diff
@@ -103,6 +104,22 @@ impl<
         context.seq = req.seq;
         context.method = req.method;
         context.user_id = self.conn_info.get_user_id();
+        context.role = self.conn_info.get_role();
+
+        // Check roles
+        let allowed_roles = self.server.allowed_roles.get(&req.method).unwrap();
```

Comment:
I'm a bit unhappy with `unwrap` as it gives no context and crashes the application. If it's okay to crash then `expect` is more preferable as it tells why a value shall be present.